wpilibc/src/main/native/cpp/controller/LTVUnicycleController.cpp

```diff
@@ -54,52 +42,40 @@ void LTVUnicycleController::SetTolerance(const Pose2d& poseTolerance) {
 }
 
 ChassisSpeeds LTVUnicycleController::Calculate(
-    const Pose2d& currentPose, const Pose2d& poseRef,
-    units::meters_per_second_t linearVelocityRef,
+    const Pose2d& currentPose, units::meters_per_second_t currentLinearVelocity,
+    const Pose2d& poseRef, units::meters_per_second_t linearVelocityRef,
     units::radians_per_second_t angularVelocityRef) {
-  Eigen::Matrix<double, 3, 1> x;
-  x(0, 0) = currentPose.Translation().X().to<double>();
-  x(1, 0) = currentPose.Translation().Y().to<double>();
-  x(2, 0) = currentPose.Rotation().Radians().to<double>();
+  m_poseError = poseRef.RelativeTo(currentPose);
 
-  Eigen::Matrix<double, 3, 1> r;
-  r(0, 0) = poseRef.Translation().X().to<double>();
-  r(1, 0) = poseRef.Translation().Y().to<double>();
-  r(2, 0) = poseRef.Rotation().Radians().to<double>();
+  if (currentLinearVelocity == 0_mps) {
+    currentLinearVelocity = 1e-9_mps;
+  } else if (units::math::abs(currentLinearVelocity) < 1e-9_mps) {
+    currentLinearVelocity = 1e-9_mps * wpi::sgn(currentLinearVelocity);
+  }
```

Comment:
There should be a comment explaining this code's existence, like this: https://github.com/frc3512/Robot-2020/blob/master/src/main/cpp/controllers/DrivetrainController.cpp#L241-L242